src/Microsoft.OData.Core/UriParser/Parsers/UriQueryExpressionParser.cs

```diff
@@ -1138,6 +1144,66 @@ private QueryToken ParsePrimaryStart()
             }
         }
 
+        private RootPathToken ParseRootPath(RootPathToken rootPathToken)
+        {
+            while (this.lexer.CurrentToken.Kind == ExpressionTokenKind.Slash)
+            {
+                this.lexer.NextToken(); // read over the '/'
+
+                // a function call or an entity with key value is treated same as function call.
+                bool identifierIsFunction = this.lexer.ExpandIdentifierAsFunction();
+                if (identifierIsFunction && TryParseIdentifierAsFunction(lexer, out string result))
+                {
+                    rootPathToken.Segments.Add(result);
+                    this.lexer.NextToken();
+                    continue;
+                }
+
+                if (this.lexer.PeekNextToken().Kind == ExpressionTokenKind.Dot)
+                {
+                    ReadOnlySpan<char> dotIdentifier = this.lexer.ReadDottedIdentifier(false);
+                    rootPathToken.Segments.Add(dotIdentifier.ToString());
+                    //lexer.NextToken();

```

Comment:
Should this be uncommented or you can add a comment on the top to explain why it is commented out.